// @archwayhq/arch3-proto/build/ibc/applications/transfer/v1/tx.amino.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AminoConverter = void 0;
const helpers_1 = require("../../../../helpers");
exports.AminoConverter = {
    "/ibc.applications.transfer.v1.MsgTransfer": {
        aminoType: "cosmos-sdk/MsgTransfer",
        toAmino: ({ sourcePort, sourceChannel, token, sender, receiver, timeoutHeight, timeoutTimestamp }) => {
            var _a, _b;
            return {
                source_port: sourcePort,
                source_channel: sourceChannel,
                token: {
                    denom: token.denom,
                    amount: helpers_1.Long.fromValue(token.amount).toString()
                },
                sender,
                receiver,
                timeout_height: timeoutHeight ? {
                    revision_height: (_a = (0, helpers_1.omitDefault)(timeoutHeight.revisionHeight)) === null || _a === void 0 ? void 0 : _a.toString(),
                    revision_number: (_b = (0, helpers_1.omitDefault)(timeoutHeight.revisionNumber)) === null || _b === void 0 ? void 0 : _b.toString()
                } : {},
                timeout_timestamp: timeoutTimestamp.toString()
            };
        },
        fromAmino: ({ source_port, source_channel, token, sender, receiver, timeout_height, timeout_timestamp }) => {
            return {
                sourcePort: source_port,
                sourceChannel: source_channel,
                token: {
                    denom: token.denom,
                    amount: token.amount
                },
                sender,
                receiver,
                timeoutHeight: timeout_height ? {
                    revisionHeight: helpers_1.Long.fromString(timeout_height.revision_height || "0", true),
                    revisionNumber: helpers_1.Long.fromString(timeout_height.revision_number || "0", true)
                } : undefined,
                timeoutTimestamp: helpers_1.Long.fromString(timeout_timestamp)
            };
        }
    }
};
