// @archwayhq/arch3-proto/build/archway/tracking/v1/genesis.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenesisState = void 0;
/* eslint-disable */
const tracking_1 = require("./tracking");
const helpers_1 = require("../../../helpers");
const _m0 = __importStar(require("protobufjs/minimal"));
function createBaseGenesisState() {
    return {
        txInfoLastId: helpers_1.Long.UZERO,
        txInfos: [],
        contractOpInfoLastId: helpers_1.Long.UZERO,
        contractOpInfos: []
    };
}
exports.GenesisState = {
    encode(message, writer = _m0.Writer.create()) {
        if (!message.txInfoLastId.isZero()) {
            writer.uint32(8).uint64(message.txInfoLastId);
        }
        for (const v of message.txInfos) {
            tracking_1.TxInfo.encode(v, writer.uint32(18).fork()).ldelim();
        }
        if (!message.contractOpInfoLastId.isZero()) {
            writer.uint32(24).uint64(message.contractOpInfoLastId);
        }
        for (const v of message.contractOpInfos) {
            tracking_1.ContractOperationInfo.encode(v, writer.uint32(34).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGenesisState();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.txInfoLastId = reader.uint64();
                    break;
                case 2:
                    message.txInfos.push(tracking_1.TxInfo.decode(reader, reader.uint32()));
                    break;
                case 3:
                    message.contractOpInfoLastId = reader.uint64();
                    break;
                case 4:
                    message.contractOpInfos.push(tracking_1.ContractOperationInfo.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            txInfoLastId: (0, helpers_1.isSet)(object.txInfoLastId) ? helpers_1.Long.fromValue(object.txInfoLastId) : helpers_1.Long.UZERO,
            txInfos: Array.isArray(object === null || object === void 0 ? void 0 : object.txInfos) ? object.txInfos.map((e) => tracking_1.TxInfo.fromJSON(e)) : [],
            contractOpInfoLastId: (0, helpers_1.isSet)(object.contractOpInfoLastId) ? helpers_1.Long.fromValue(object.contractOpInfoLastId) : helpers_1.Long.UZERO,
            contractOpInfos: Array.isArray(object === null || object === void 0 ? void 0 : object.contractOpInfos) ? object.contractOpInfos.map((e) => tracking_1.ContractOperationInfo.fromJSON(e)) : []
        };
    },
    toJSON(message) {
        const obj = {};
        message.txInfoLastId !== undefined && (obj.txInfoLastId = (message.txInfoLastId || helpers_1.Long.UZERO).toString());
        if (message.txInfos) {
            obj.txInfos = message.txInfos.map(e => e ? tracking_1.TxInfo.toJSON(e) : undefined);
        }
        else {
            obj.txInfos = [];
        }
        message.contractOpInfoLastId !== undefined && (obj.contractOpInfoLastId = (message.contractOpInfoLastId || helpers_1.Long.UZERO).toString());
        if (message.contractOpInfos) {
            obj.contractOpInfos = message.contractOpInfos.map(e => e ? tracking_1.ContractOperationInfo.toJSON(e) : undefined);
        }
        else {
            obj.contractOpInfos = [];
        }
        return obj;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseGenesisState();
        message.txInfoLastId = object.txInfoLastId !== undefined && object.txInfoLastId !== null ? helpers_1.Long.fromValue(object.txInfoLastId) : helpers_1.Long.UZERO;
        message.txInfos = ((_a = object.txInfos) === null || _a === void 0 ? void 0 : _a.map(e => tracking_1.TxInfo.fromPartial(e))) || [];
        message.contractOpInfoLastId = object.contractOpInfoLastId !== undefined && object.contractOpInfoLastId !== null ? helpers_1.Long.fromValue(object.contractOpInfoLastId) : helpers_1.Long.UZERO;
        message.contractOpInfos = ((_b = object.contractOpInfos) === null || _b === void 0 ? void 0 : _b.map(e => tracking_1.ContractOperationInfo.fromPartial(e))) || [];
        return message;
    }
};
