// @archwayhq/arch3-proto/build/ibc/core/connection/v1/tx.amino.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AminoConverter = void 0;
const helpers_1 = require("../../../../helpers");
exports.AminoConverter = {
    "/ibc.core.connection.v1.MsgConnectionOpenInit": {
        aminoType: "cosmos-sdk/MsgConnectionOpenInit",
        toAmino: ({ clientId, counterparty, version, delayPeriod, signer }) => {
            return {
                client_id: clientId,
                counterparty: {
                    client_id: counterparty.clientId,
                    connection_id: counterparty.connectionId,
                    prefix: {
                        key_prefix: counterparty.prefix.keyPrefix
                    }
                },
                version: {
                    identifier: version.identifier,
                    features: version.features
                },
                delay_period: delayPeriod.toString(),
                signer
            };
        },
        fromAmino: ({ client_id, counterparty, version, delay_period, signer }) => {
            return {
                clientId: client_id,
                counterparty: {
                    clientId: counterparty.client_id,
                    connectionId: counterparty.connection_id,
                    prefix: {
                        keyPrefix: counterparty.prefix.key_prefix
                    }
                },
                version: {
                    identifier: version.identifier,
                    features: version.features
                },
                delayPeriod: helpers_1.Long.fromString(delay_period),
                signer
            };
        }
    },
    "/ibc.core.connection.v1.MsgConnectionOpenTry": {
        aminoType: "cosmos-sdk/MsgConnectionOpenTry",
        toAmino: ({ clientId, previousConnectionId, clientState, counterparty, delayPeriod, counterpartyVersions, proofHeight, proofInit, proofClient, proofConsensus, consensusHeight, signer }) => {
            var _a, _b, _c, _d;
            return {
                client_id: clientId,
                previous_connection_id: previousConnectionId,
                client_state: {
                    type_url: clientState.typeUrl,
                    value: clientState.value
                },
                counterparty: {
                    client_id: counterparty.clientId,
                    connection_id: counterparty.connectionId,
                    prefix: {
                        key_prefix: counterparty.prefix.keyPrefix
                    }
                },
                delay_period: delayPeriod.toString(),
                counterparty_versions: counterpartyVersions.map(el0 => ({
                    identifier: el0.identifier,
                    features: el0.features
                })),
                proof_height: proofHeight ? {
                    revision_height: (_a = (0, helpers_1.omitDefault)(proofHeight.revisionHeight)) === null || _a === void 0 ? void 0 : _a.toString(),
                    revision_number: (_b = (0, helpers_1.omitDefault)(proofHeight.revisionNumber)) === null || _b === void 0 ? void 0 : _b.toString()
                } : {},
                proof_init: proofInit,
                proof_client: proofClient,
                proof_consensus: proofConsensus,
                consensus_height: consensusHeight ? {
                    revision_height: (_c = (0, helpers_1.omitDefault)(consensusHeight.revisionHeight)) === null || _c === void 0 ? void 0 : _c.toString(),
                    revision_number: (_d = (0, helpers_1.omitDefault)(consensusHeight.revisionNumber)) === null || _d === void 0 ? void 0 : _d.toString()
                } : {},
                signer
            };
        },
        fromAmino: ({ client_id, previous_connection_id, client_state, counterparty, delay_period, counterparty_versions, proof_height, proof_init, proof_client, proof_consensus, consensus_height, signer }) => {
            return {
                clientId: client_id,
                previousConnectionId: previous_connection_id,
                clientState: {
                    typeUrl: client_state.type_url,
                    value: client_state.value
                },
                counterparty: {
                    clientId: counterparty.client_id,
                    connectionId: counterparty.connection_id,
                    prefix: {
                        keyPrefix: counterparty.prefix.key_prefix
                    }
                },
                delayPeriod: helpers_1.Long.fromString(delay_period),
                counterpartyVersions: counterparty_versions.map(el0 => ({
                    identifier: el0.identifier,
                    features: el0.features
                })),
                proofHeight: proof_height ? {
                    revisionHeight: helpers_1.Long.fromString(proof_height.revision_height || "0", true),
                    revisionNumber: helpers_1.Long.fromString(proof_height.revision_number || "0", true)
                } : undefined,
                proofInit: proof_init,
                proofClient: proof_client,
                proofConsensus: proof_consensus,
                consensusHeight: consensus_height ? {
                    revisionHeight: helpers_1.Long.fromString(consensus_height.revision_height || "0", true),
                    revisionNumber: helpers_1.Long.fromString(consensus_height.revision_number || "0", true)
                } : undefined,
                signer
            };
        }
    },
    "/ibc.core.connection.v1.MsgConnectionOpenAck": {
        aminoType: "cosmos-sdk/MsgConnectionOpenAck",
        toAmino: ({ connectionId, counterpartyConnectionId, version, clientState, proofHeight, proofTry, proofClient, proofConsensus, consensusHeight, signer }) => {
            var _a, _b, _c, _d;
            return {
                connection_id: connectionId,
                counterparty_connection_id: counterpartyConnectionId,
                version: {
                    identifier: version.identifier,
                    features: version.features
                },
                client_state: {
                    type_url: clientState.typeUrl,
                    value: clientState.value
                },
                proof_height: proofHeight ? {
                    revision_height: (_a = (0, helpers_1.omitDefault)(proofHeight.revisionHeight)) === null || _a === void 0 ? void 0 : _a.toString(),
                    revision_number: (_b = (0, helpers_1.omitDefault)(proofHeight.revisionNumber)) === null || _b === void 0 ? void 0 : _b.toString()
                } : {},
                proof_try: proofTry,
                proof_client: proofClient,
                proof_consensus: proofConsensus,
                consensus_height: consensusHeight ? {
                    revision_height: (_c = (0, helpers_1.omitDefault)(consensusHeight.revisionHeight)) === null || _c === void 0 ? void 0 : _c.toString(),
                    revision_number: (_d = (0, helpers_1.omitDefault)(consensusHeight.revisionNumber)) === null || _d === void 0 ? void 0 : _d.toString()
                } : {},
                signer
            };
        },
        fromAmino: ({ connection_id, counterparty_connection_id, version, client_state, proof_height, proof_try, proof_client, proof_consensus, consensus_height, signer }) => {
            return {
                connectionId: connection_id,
                counterpartyConnectionId: counterparty_connection_id,
                version: {
                    identifier: version.identifier,
                    features: version.features
                },
                clientState: {
                    typeUrl: client_state.type_url,
                    value: client_state.value
                },
                proofHeight: proof_height ? {
                    revisionHeight: helpers_1.Long.fromString(proof_height.revision_height || "0", true),
                    revisionNumber: helpers_1.Long.fromString(proof_height.revision_number || "0", true)
                } : undefined,
                proofTry: proof_try,
                proofClient: proof_client,
                proofConsensus: proof_consensus,
                consensusHeight: consensus_height ? {
                    revisionHeight: helpers_1.Long.fromString(consensus_height.revision_height || "0", true),
                    revisionNumber: helpers_1.Long.fromString(consensus_height.revision_number || "0", true)
                } : undefined,
                signer
            };
        }
    },
    "/ibc.core.connection.v1.MsgConnectionOpenConfirm": {
        aminoType: "cosmos-sdk/MsgConnectionOpenConfirm",
        toAmino: ({ connectionId, proofAck, proofHeight, signer }) => {
            var _a, _b;
            return {
                connection_id: connectionId,
                proof_ack: proofAck,
                proof_height: proofHeight ? {
                    revision_height: (_a = (0, helpers_1.omitDefault)(proofHeight.revisionHeight)) === null || _a === void 0 ? void 0 : _a.toString(),
                    revision_number: (_b = (0, helpers_1.omitDefault)(proofHeight.revisionNumber)) === null || _b === void 0 ? void 0 : _b.toString()
                } : {},
                signer
            };
        },
        fromAmino: ({ connection_id, proof_ack, proof_height, signer }) => {
            return {
                connectionId: connection_id,
                proofAck: proof_ack,
                proofHeight: proof_height ? {
                    revisionHeight: helpers_1.Long.fromString(proof_height.revision_height || "0", true),
                    revisionNumber: helpers_1.Long.fromString(proof_height.revision_number || "0", true)
                } : undefined,
                signer
            };
        }
    }
};
